robottelo/vm.py

```diff
@@ -257,6 +257,82 @@ def run(self, cmd):
 
         return ssh.command(cmd, hostname=self.ip_addr)
 
+    def fetch_hostname(self):
+        """Fetches short hostname from client and the domain from provisioning
+        server.
+
+        :return: Returns the FQDN of the host.
+
+        """
+        # NOTE: We will be using 'hostname -s' as currently the hostname being
+        # returned from virtual machine is short hostname. It is being done to
+        # handle the scenario where in the vm/client might return FQDN for
+        # a different provisioning server.
+        host = self.run('hostname -s')
+        domain = ssh.command(
+            u'hostname -d',
+            hostname=self.provisioning_server
+        )
+        return host.stdout[0] + u'.' + domain.stdout[0]
+
+    def configure_rhel_repo(self, rhel_repo):
+        """Configures specified Red Hat repository on the virtual machine.
+
+        :param rhel_repo: Red Hat repository link from properties file.
+        :return: None.
+
+        """
+        # 'Access Insights', 'puppet' requires RHEL 6/7 repo and it is not
+        # possible to sync the repo during the tests as they are huge(in GB's)
+        # hence this adds a file in /etc/yum.repos.d/rhel6/7.repo
+        repo_file = (
```

Comment:
I think depending upon the provisioning server's location, the "rhel6-baseurl" link could change for Brno or pune or rdu. So I think we should avoid rhel_repo. 

download.devel.redhat.com for US folks
downloda.end.pnq.redhat.com for pnq folks as it will be on LAN.

I would prefer using pnq as it would be on LAN if I tomorrow plan to use local provisioning_Server other than ginger.

This would help have more control over the automation for ppl in different geographic location and using provisioning_Server other than "ginger".